Compute the preview canvas size from the rendered elements

jersub opened the pull request:
Fixes issue #572 

For each item to render on a Preview target, the canvas size is computed. Then, from the union of all of these rectangles, one gets the global Preview canvas size.

With such an implementation, any Preview plugin can declare the canvas size of its own items to render, so they can be integrated in the global Preview canvas size just like the regular items.

mbastian commented on modules/PreviewAPI/src/main/java/org/gephi/preview/PreviewModelImpl.java:
```diff
@@ -82,8 +80,7 @@ Development and Distribution License("CDDL") (collectively, the
     //Dimensions
-    private Dimension dimensions;
-    private Point topLeftPosition;
+    private Rectangle _graphicsCanvas;
```
We don't use the _ convention for class variables, should be removed.

jersub replied:
I put it to try to tell developers not to use this variable directly, `getGraphicsCanvas()` has to be used instead as it defines the variable. Any suggestion for handling this?

mbastian replied:
Add a comment instead, this is implementation code (not API) so just use comments

jersub replied:
Actually it was a mistake to cache the canvas size. It has to be computed each time to get the last info from the graph model.

mbastian commented on modules/PreviewAPI/src/main/java/org/gephi/preview/api/Rectangle.java:
```diff
@@ -0,0 +1,148 @@
+ *
+ * @author Jeremy Subtil
+ */
+public class Rectangle implements Shape {
```
I think adding a Rectangle class to the API is confusing. Users don't need any new feature besides the position/width so it shouldn't be in the API. The union is used internally by the `PreviewModelImpl` and can be called in a different way than by subclassing Rectangle.

jersub commented:
Thanks a lot for the review, I'll handle your comments.

jersub commented:
Done. Sorry for the huge diff again, I amended my previous commit too quickly.

Brief recap: replaced `Rectangle` by `CanvasSize`, removed the canvas size cache from `PreviewModelImpl`.

mbastian commented:
Ok, thank you, I'll take a look at it tomorrow morning. Did you notice any net performance issues with large graphs? Does the canvas calculation also happen for the Preview Panel (Java2D) or is it only for export? Thanks. Just curious before reading the code :)

jersub commented:
I haven't done any real performance check for now. The canvas calculation does also happen to the Preview Panel.

jersub commented:
Execution time for `getGraphicsCanvasSize()` on a big graph: < 30 ms

Called on a first Preview refresh, a zoom reset and an export.

mbastian commented:
Awesome, thanks for checking! I saw that you left the label calculation for later, this is fine but curious whether it can easily be added. Merging nevertheless.
